src/core/main.c

```diff
@@ -2404,6 +2411,13 @@ static void reset_arguments(void) {
         arg_clock_usec = 0;
 
         arg_default_oom_score_adjust_set = false;
+#ifdef SMACK_DEFAULT_PROCESS_LABEL
+        if (free_and_strdup(arg_default_smack_process_label,
+                            SMACK_DEFAULT_PROCESS_LABEL) < 0)
+                log_oom();
```

Comment:
reset_arguments() is not supposed to fail...

I think it would be better if arg_default_smack_process_label would always default to NULL, but then, when you copy it into Manager you fill in SMACK_DEFAULT_PROCESS_LABEL if still NULL.

that way you remove one pointless string copy, and don't have the reset logic here fail, and things would generally be simpler